diesel/src/connection/mod.rs

```diff
@@ -39,8 +39,15 @@ pub trait ConnectionGatWorkaround<'a, DB: Backend> {
     type Row: crate::row::Row<'a, DB>;
 }
 
+/// Methods necessary for the transaction manager
+pub trait TransactionalConnection {
```

Comment:
I feel that adding a whole new trait for this is to complicated. I would like to see possible alternatives here. Optimally this should be handled inside of the transaction manager implementation only.

Reply:
@weiznich , relying on localized error messages is really fragile and [discouraged](https://www.postgresql.org/docs/14/errcodes-appendix.html) by the Postgres documentation:

> All messages emitted by the PostgreSQL server are assigned five-character error codes that follow the SQL standard's conventions for “SQLSTATE” codes. Applications that need to know which error condition has occurred should usually test the error code, rather than looking at the textual error message. The error codes are less likely to change across PostgreSQL releases, and also are not subject to change due to localization of error messages. Note that some, but not all, of the error codes produced by PostgreSQL are defined by the SQL standard; some additional error codes for conditions not defined by the standard have been invented or borrowed from other databases.

Forcing the error messages to be in English would also be potentially a breaking change for Diesel consumers, without compilation errors.

I can suggest two approaches if you don't like the additional trait:
- move `is_transaction_broken()` to the `Connection` trait
- strongly-typed "raw" error codes in `Error`, ie `RawError::Postgres(SomePostgresErrorCodeEnum)` - this would be a good idea in general, but it's also a lot more work

Reply:
As a side note, do you know why the CI tests have no output? It's somewhat disturbing

Reply:
We could also change the bounds like so:
`trait TransactionalConnection: Connection`, and use `TransactionalConnection` instead of `Connection` as bound for `AnsiTransactionManager`'s parameter. This would make a lot of sense, and `TransactionalConnection`' implementations could move closer to `AnsiTransactionManager`.